cdap-watchdog/src/main/java/co/cask/cdap/logging/gateway/handlers/LogDataOffsetProducer.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.logging.gateway.handlers;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.classic.spi.ThrowableProxyUtil;
+import co.cask.cdap.api.dataset.lib.CloseableIterator;
+import co.cask.cdap.logging.read.LogEvent;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import java.util.List;
+
+/**
+ * LogReader BodyProducer to encode log events, as {@link LogData} objects.
+ */
+public class LogDataOffsetProducer extends AbstractJSONLogProducer {
+  private final List<String> fieldsToSuppress;
+
+  LogDataOffsetProducer(CloseableIterator<LogEvent> logEventIter, List<String> fieldsToSuppress) {
+    super(logEventIter);
+    this.fieldsToSuppress = fieldsToSuppress;
+    validate();
+  }
+
+  @Override
+  public Object encodeSend(LogEvent logEvent) {
```

Comment:
From performance standpoint, it's better to have encoding done using a Gson `TypeAdapter` instead of converting `LogEvent` to another POJO `Object`.

No need to do it for this PR, but I thing it's worth refactoring in future.